src/azure-cli/azure/cli/command_modules/cdn/custom/custom.py

```diff
@@ -422,14 +446,34 @@ def add_action(client, resource_group_name, profile_name, endpoint_name,
     return client.endpoints.begin_update(resource_group_name, profile_name, endpoint_name, params)
 
 
-def remove_rule(client, resource_group_name, profile_name, endpoint_name, rule_name):
+def remove_rule(client, resource_group_name, profile_name, endpoint_name, rule_name=None, order: int = None):
+
+    if rule_name is None and order is None:
+        raise CLIError("Either --rule-name or --order must be specified")
+
+    if order is not None and order < 0:
+        raise CLIError("Order should be non-negative.")
 
     endpoint = client.endpoints.get(resource_group_name, profile_name, endpoint_name)
     policy = endpoint.delivery_policy
     if policy is not None:
-        for rule in policy.rules:
-            if rule.name == rule_name:
-                policy.rules.remove(rule)
+        _convert_to_unified_delivery_rules(policy)
+        pop_index = -1
+        for idx, rule in enumerate(policy.rules):
+            if rule_name is not None and rule.name == rule_name:
+                pop_index = idx
+                break
+            elif order is not None and rule.order == order:
+                pop_index = idx
+                break
+
+        if pop_index != -1:
```

Comment:
Could we add some comments so that reviewers can know why we need this change?

Reply:
Will add some comments